src/components/VideoContainer.jsx

```diff
@@ -1,35 +1,44 @@
-import { useParams } from 'react-router-dom';
-import { useContext } from 'react';
+import { youtubeVideoUrl } from '../utils/constants';
+import { useContext, useState, useEffect } from 'react';
 import { GlobalContext } from '../context/GlobalContext';
 import { StyledVideoContainer } from './styles/VideoContainer.styled';
 import { lightStyles } from './styles/ThemeStyles.styled';
 import { darkStyles } from './styles/ThemeStyles.styled';
 import { StyledVideoTitle } from './styles/StyledVideoTitle.styled';
 import { StyledVideoDescription } from './styles/StyledVideoDescription.styled';
-import { youtubeVideoUrl } from '../utils/constants';
-import { youtubeApiUrl } from '../utils/constants';
-import useFetch from '../hooks/useFetch';
 import Alert from 'react-bootstrap/Alert';
+import Button from 'react-bootstrap/Button';
 import Spinner from './Spinner';
 import ReactPlayer from 'react-player';
+import PropTypes from 'prop-types';
+
+const VideoContainer = ({ loading, error, video, id, favoritesPage }) => {
+  const { darkMode, isLoggedIn, favorites, dispatch } =
+    useContext(GlobalContext);
 
-const VideoContainer = () => {
-  const { darkMode } = useContext(GlobalContext);
   const textColor = darkMode ? darkStyles.color : lightStyles.color;
-  const { id } = useParams();
-  const { loading, error, data } = useFetch(
-    // eslint-disable-next-line no-undef
-    `${youtubeApiUrl}videos?part=snippet&id=${id}&key=${process.env.REACT_APP_API_KEY}`,
-    {},
-    id
-  );
+
   if (error) {
     return <Alert />;
   }
-  const videoTitle = data ? data.items[0].snippet.localized.title : '';
-  const videoDescription = data
-    ? data.items[0].snippet.localized.description
-    : '';
+
+  const [isVideoInFavorites, setIsVideoInFavorites] = useState(null);
+
+  useEffect(() => {
+    setIsVideoInFavorites(favorites.some((favorite) => favorite.id === id));
+  }, [id]);
```

Comment:
I think the best thing would be that instead of handling this logic directly here would be great if you find a way to add the video data a flag that indicates if it is in favorites or not. and then when you dispatch the action also add or remove the flag there